editors/emacs/lambdapi-mode.el

```diff
@@ -0,0 +1,143 @@
+(require 'eglot)
+;;; Legacy
+;; Syntax table (legacy syntax)
+(defvar lambdapi-mode-legacy-syntax-table nil "Syntax table for LambdaPi.")
+
+(setq lambdapi-mode-legacy-syntax-table
+  (let ((syn-table (make-syntax-table)))
+    (modify-syntax-entry ?\( "()1n" syn-table)
+    (modify-syntax-entry ?\) ")(4n" syn-table)
+    (modify-syntax-entry ?\; ". 23" syn-table)
+    syn-table))
+
+;; Keywords (legacy syntax)
+(defconst lambdapi-legacy-font-lock-keywords
+  (list
+   (cons
+    (concat
+     "\\<"
+     (regexp-opt '("def" "thm" "inj"))
+     "\\>") 'font-lock-keyword-face)
+   (cons
+    (concat
+     "#"
+     (regexp-opt '("REQUIRE" "EVAL" "INFER" "ASSERT" "ASSERTNOT"))
+     "\\>") 'font-lock-preprocessor-face))
+  "Keyword highlighting for the LambdaPi mode (legacy syntax).")
+
+
+;; Main function creating the mode (legacy syntax)
+;;;###autoload
+(define-derived-mode lambdapi-legacy-mode prog-mode "LambdaPi (legacy)"
+  "A mode for editing LambdaPi files (in legacy syntax)."
+  (set-syntax-table lambdapi-mode-legacy-syntax-table)
+  (setq-local font-lock-defaults '(lambdapi-legacy-font-lock-keywords))
+  (setq-local comment-start "(;")
+  (setq-local comment-end ";)")
+  (setq-default indent-tabs-mode nil)
+  (add-to-list 'eglot-server-programs
+               '(lambdapi-legacy-mode . ("lambdapi" "lsp" "--standard-lsp")))
+  (add-hook 'lambdapi-legacy-mode-hook 'eglot-ensure))
+
+(provide 'lambdapi-legacy-mode)
+
+;;; lambdapi
+;; Keywords
+(defconst lambdapi-font-lock-keywords
+  (list (cons
+         (concat "\\<" (regexp-opt lambdapi-sig-commands) "\\>")
+         'font-lock-keyword-face)
+        (cons
+         (concat "\\<" (regexp-opt lambdapi-misc-commands) "\\>")
+         'font-lock-preprocessor-face)
+        (cons
+         (concat "\\<" (regexp-opt lambdapi-modifiers) "\\>")
+         'font-lock-warning-face)
+        (cons
+         (concat "\\<" (regexp-opt lambdapi-tactics) "\\>")
+         'font-lock-builtin-face)
+        (cons
+         (concat "\\<" (regexp-opt lambdapi-misc-keywords) "\\>")
+         'font-lock-constant-face))
+  "Keyword highlighting for the LambdaPi mode.")
+
+;; Unicode support
+(require 'quail)
+(quail-define-package
+  "LambdaPi" "LambdaPi" "LambdaPi" t
+  "A transliteration scheme for LambdaPi."
+  nil t t t t nil t nil nil nil t)
+(quail-define-rules
+  ("->" ?→) ("\\to" ?→) ("\\rightarrow" ?→)
+  ("=>" ?⇒) ("\\To" ?⇒) ("\\Rightarrow" ?⇒)
+  ("!!" ?∀) ("\\forall" ?∀)
+  ("\\\\" ?λ) ("\\lambda" ?λ)
+  (":=" ?≔) ("\\defeq"  ?≔)
+  ("==" ?≡) ("\\equiv"  ?≡)
+  ("&&" ?∧) ("\\wedge"  ?∧) ("/\\" ?∧)
+  ("||" ?∨) ("\\vee"    ?∨) ("\\/" ?∨)
+  ("~~" ?¬) ("\\neg"    ?¬)
+  ("??" ?∃) ("\\exists" ?∃)
+  ("`a" ?α) ("`b" ?β) ("`c" ?γ) ("`d" ?δ)
+  ("`e" ?ε) ("`h" ?η) ("`i" ?ι) ("`k" ?κ)
+  ("`m" ?μ) ("`n" ?ν) ("`o" ?ω) ("`p" ?π)
+  ("`r" ?ρ) ("`s" ?σ) ("`t" ?τ) ("`u" ?υ)
+  ("`w" ?ω) ("`x" ?χ) ("`y" ?υ) ("`z" ?ζ)
+  ("`C" ?Γ) ("`D" ?Δ) ("`G" ?Γ) ("`L" ?Λ)
+  ("`O" ?Ω) ("`P" ?Π) ("`S" ?Σ) ("`W" ?Ω))
```

Comment:
This doesn't seem very standard. Why not converting \alpha into α, etc. People are more used to Latex commands. Isn't there already a mode doing this?

Reply:
Well, it's standard if you use `cdlatex`... but I see what you mean, it is done this way in the Agda mode,
<https://github.com/agda/agda/blob/323f58f9b8dad239142ed1dfa0c60338ea2cb157/src/data/emacs-mode/agda-input.el>

Reply:
`cdlatex` is quite standard for emacs, and I personally find it convenient.

Reply:
And no, it seems there are no mode doing it (at least Agda hasn't found any mode either). But there is the TeX input method (`C-x RET C-\` or `M-x set-input-method` to try it). One possible option would be to merge the TeX input method with the LambdaPi one

Reply:
What do you think of the TeX input mode? See https://github.com/emacs-mirror/emacs/blob/master/lisp/leim/quail/latin-ltx.el . Could you activate it for LP files? I also found https://github.com/Christoph-D/LaTeX-unicode-math-mode.

Reply:
I initially thought it would be too much, but after all, why not. Your second suggestion is made for TeX modes, and needs the strings to be in math mode (so inside `$`s)

Reply:
This is not what says https://github.com/Christoph-D/LaTeX-unicode-math-mode

Reply:
Better: https://github.com/vspinu/company-math
```
(add-hook 'lambdapi-mode-hook (lambda () (setq-local company-backends '((company-math-symbols-unicode)))))
```

Reply:
For the company mode, I tried it, but without success (I can't figure out why); and it's more up to the user to configure this.
Indeed, I misread, but I found a satisfactory solution relying on
<https://elpa.gnu.org/packages/math-symbol-lists.html>.

As of c4a2470, all greek letters are available via their latex command.

Reply:
With https://github.com/Deducteam/lambdapi/pull/308/commits/c4a24709b661882c810d164338783ea396e34684 my previous hook doesn't work anymore!

Reply:
@gabrielhdt could you please make your new emacs mode use https://github.com/vspinu/company-math ?

Reply:
@fblanqui, you could before? I never managed to get this company thing working. Can you get what you want using alpha input, though?

Reply:
I don't understand. It doesn't work anymore even if I go back to master.
Question: with the new emacs mode, do we need to write (load "lambdapi") in our ~/.emacs?

Reply:
What does not work anymore? The new emacs mode does not require any `(load "lamdbapi")` line, when the package is installed, some functions are auto (and lazy) loaded into emacs. To remove the package, `package-delete RET lambdapi-mode-0.1` should do the trick.

Reply:
First, if I do not add (load "lambdapi") in ~/.emacs, I get no lambdapi mode when I open a LP file.

Reply:
It's normal if you removed my package or haven't installed it